openchrom/plugins/net.openchrom.nmr.process.supplier.knime.ui/src/net/openchrom/nmr/process/supplier/knime/ui/reader/ReaderNMRNodeModel.java

```diff
@@ -84,8 +85,8 @@ protected ReaderNMRNodeModel() {
 		File file = new File(settingsFileInput.getStringValue());
 		try {
 			IProcessingInfo processingInfo = ScanConverterNMR.convert(file, new NullProgressMonitor());
-			IDataNMRSelection scanNMR = (IDataNMRSelection)processingInfo.getProcessingResult();
-			return new PortObject[]{new ScanNMRPortObject(scanNMR)};
+			MeasurementNMR scanNMR = (MeasurementNMR)processingInfo.getProcessingResult();
```

Comment:
I think the result should not be just casted to any interface but type checked via "instanceof", I have seen various objects placed as the "result" even exceptions...

Reply:
Yes, imho a check like
`Object object = processingInfo.getProcessingResult();
if(object instanceof IMeasurementNMR) {
// do processing
}`
will bemore reliable.

Reply:
No, I dont think so. Better to get an runtime error to see there is a problem instead of getting no error at all and having a hard time to find out why nothing is happening.

Reply:
ClassCastException is always bad, because hard to understand, hard to debug and could be avoided with proper checks.
If it is of wrong type you have two options:
* throw a meaningful exception (e.g. "processing has returned an unexpected result of type XYZ maybe the file input is of wrong format or whatever can cause this)"
* choose some other kind of feedback, e.g showing an error marker or something, I don't know what is the expected way in KNIME to do so if node processing failed

Reply:
It might be good to return for example an error port object showing the error inside a TextField

Reply:
ClassCastException will clearly indicate a programming error. As long as processingResult is not generic, ClassCastException is just fine, because its just a symptom of another problem.